### DISCUSSION TITLE
Differences between external library and read only gallery

### DISCUSSION THREAD

**Discussion by asonu0917** [immich-app/immich#4161] [Q&A] [answered]
I am trying to have my existing photos show in the application, but without re-uploading, or moving them. New photos will also upload to that location instead of using Immich auto backup feature.

While reading the documents, I came across 2 different approaches those seems fit my needs. However, they look very similar to me. Anyone can help to explain the differences and give some advice  on these 2 features?

I understand Read-only Gallery required CLI to trigger the import, while external library seems to have UI components to aid the process (not sure if it can be scheduled).

https://documentation.immich.app/docs/features/read-only-gallery
https://documentation.immich.app/docs/features/libraries#external-libraries

**Comment by bo0tzz** [comment, marked as answer]
Library support has just been merged. Once it's mature, it's probable that it will replace the read-only gallery entirely.
